ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationPeriodValidationIT.scala

```diff
@@ -161,30 +162,76 @@ class CommandDeduplicationPeriodValidationIT extends LedgerTestSuite {
     disabledReason = "The ledger does not support deduplication periods represented by offsets",
     runConcurrently = false, // Pruning is involved
   )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    def submitAndWaitWithDeduplication(
+        deduplicationPeriod: DeduplicationPeriod.DeduplicationOffset
+    ) = {
+      ledger
+        .submitAndWait(
+          ledger
+            .submitAndWaitRequest(party, Dummy(party).create.command)
+            .update(
+              _.commands.deduplicationPeriod := deduplicationPeriod
+            )
+        )
+    }
+    val isOffsetNativlySupported =
+      ledger.features.commandDeduplicationFeatures.getDeduplicationPeriodSupport.offsetSupport.isOffsetNativeSupport
     for {
+      start <- ledger.currentEnd()
       firstCreate <- ledger.create(party, Dummy(party))
-      firstExercise <- ledger.exercise(party, firstCreate.exerciseDummyChoice1)
-      end <- ledger.currentEnd()
+      _ <- ledger.exercise(party, firstCreate.exerciseDummyChoice1)
       secondCreate <- ledger.create(party, Dummy(party))
+      _ <- ledger.submitAndWait(ledger.submitAndWaitRequest(party, Dummy(party).create.command))
+      end <- ledger.currentEnd()
       _ <- ledger.exercise(party, secondCreate.exerciseDummyChoice1)
       _ <- ledger.create(party, Dummy(party)) // move ledger end
+      _ <- ledger.submitAndWait(ledger.submitAndWaitRequest(party, Dummy(party).create.command))
       _ <- ledger.prune(pruneUpTo = end)
-      deduplicationPeriod = DeduplicationPeriod.DeduplicationOffset(
-        Ref.HexString.assertFromString(firstExercise.offset)
-      )
-      _ <- assertFailedRequest(
+      failure <- submitAndWaitWithDeduplication(
+        DeduplicationPeriod.DeduplicationOffset(
+          start.getAbsolute
+        )
+      ).mustFail("using an offset which was pruned")
+      _ = assertGrpcErrorRegex(
         ledger,
-        party,
-        deduplicationPeriod,
-        failReason = "Submitting a command with a pruned offset",
-        expectedMessage = ".*",
-        expectedCode = Status.Code.INVALID_ARGUMENT,
-        expectedError = LedgerApiErrors.RequestValidation.ParticipantPrunedDataAccessed,
+        failure,
+        expectedCode = Status.Code.FAILED_PRECONDITION,
+        // Canton returns INVALID_DEDUPLICATION_PERIOD with earliest_offset metadata
+        // KV returns PARTICIPANT_PRUNED_DATA_ACCESSED with earliest_offset metadata
+        selfServiceErrorCode =
+          if (isOffsetNativlySupported)
+            LedgerApiErrors.RequestValidation.InvalidDeduplicationPeriodField
+          else LedgerApiErrors.RequestValidation.ParticipantPrunedDataAccessed,
```

Comment:
If either error code is valid, I would like to see something like:

```suggestion
        // either is valid
        selfServiceErrorCode = Seq(
          LedgerApiErrors.RequestValidation.InvalidDeduplicationPeriodField,
          LedgerApiErrors.RequestValidation.ParticipantPrunedDataAccessed,
        )
```

Reply:
Well not really. Ideally, we should move to have only `InvalidDeduplicationPeriodField` but that's not currently possible because of the inclusive nature for offsets in KV.